sdks/java/io/solr/src/main/java/org/apache/beam/sdk/io/solr/SolrIO.java

```diff
@@ -364,36 +372,41 @@ Read withBatchSize(int batchSize) {
       // by tuning batchSize when pipelines run.
       checkArgument(
           batchSize > 0 && batchSize < MAX_BATCH_SIZE,
-          "Valid values for batchSize are 1 (inclusize) to %s (exclusive), but was: %s ",
+          "Valid values for batchSize are 1 (inclusive) to %s (exclusive), but was: %s ",
           MAX_BATCH_SIZE,
           batchSize);
       return builder().setBatchSize(batchSize).build();
     }
 
+    /** Read from a specific Replica (partition). */
+    public Read withReplicaInfo(ReplicaInfo replicaInfo) {
+      checkArgument(replicaInfo != null, "replicaInfo can not be null");
+      return builder().setReplicaInfo(replicaInfo).build();
+    }
+
     @Override
     public PCollection<SolrDocument> expand(PBegin input) {
       checkArgument(
           getConnectionConfiguration() != null, "withConnectionConfiguration() is required");
       checkArgument(getCollection() != null, "from() is required");
-
-      return input
-          .apply("Create", Create.of(this))
-          .apply("Split", ParDo.of(new SplitFn()))
-          .apply("Reshuffle", Reshuffle.viaRandomKey())
-          .apply("Read", ParDo.of(new ReadFn()));
+      return input.apply("Create", Create.of(this)).apply("ReadAll", readAll());
```

Comment:
Is `withReplicaInfo()` required here? If yes, please, add a check here.

Reply:
My bad, I think we don't it here since anyway it will pass through `SplitFn` downstream